File: library.py
class Library:
    """This class represents a simaris target and is initialized with
    data in JSON format

    """

    def __init__(self, data):
        if 'CurrentTarget' in data:
            self.target          = data['CurrentTarget']['EnemyType']
            self.scans           = data['CurrentTarget']['PersonalScansRequired']
            self.progress        = data['CurrentTarget']['ProgressPercent']
            self.active = True
        else:
            self.active = False

    def __str__(self):
        """Returns a string with all the information about this alert

        """
        if not self.is_active():
            return None

        library_string = ('Target: {0}\n'
                       'Scans needed: {1}\n'
                       'Progress: {2:.2f}%'
                       )

        return library_string.format(self.target, self.scans,
                                  self.progress)

    def is_active(self):
        """ Returns True if there is a currently active target, False otherwise

        """
        return self.active

File: scripts/lib/library.js
util = require('util');

var Library = function(data) {
  if(data.CurrentTarget) {
    this.active = true;
    this.target = data.CurrentTarget.EnemyType;
    this.scans = data.CurrentTarget.PersonalScansRequired;
    this.progress = data.CurrentTarget.ProgressPercent;
  }
  else {
    this.active = false;
  }
}

Library.prototype.toString = function() {
  if(!this.isActive()) {
    return 'No active target';
  }

  var libraryString = util.format('Target: %s\n' +
                                  'Scans needed: %d\n' +
                                  'Progress: %d', this.target, this.scans,
                                  Math.round(100 * this.progress) / 100);

  return libraryString;
}

Library.prototype.isActive = function() {
  return this.active;
}

module.exports = Library;
